server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -371,6 +401,19 @@ public long getInterruptedQueryCount()
     return interruptedQueryCount.get();
   }
 
+  private static byte[] getRequestContent(HttpServletRequest request) throws IOException
+  {
+    int contentSize = request.getContentLength();
```

Comment:
HTTP requests aren't required to specify a content length (for example if they use chunked encoding). So this isn't the right way to figure out how long a request is. See the spec for the getContentLength method: it may return -1.